### PR TITLE
Refactoring e2e test - fixing waiting period when delete db files

### PR DESCRIPTION
**Pull request by OmerBenHayun**
Signed-off-by: Omer Ben-Hayun Omer.Ben-Hayun@ibm.com

**- What this PR does and why is it needed**

this PR refactor the e2e test `"recovering from deleting db files while maintain connectivity"` and instead of waiting fixed amount of time (90 seconds) for the pod to restart, it will wait for the pod to finish full restart with `waitForPodToFinishFullRestart`.
the reason for the change is that  its not guaranteed that the deleted db-pod will be up and running by 90s.
**- Special notes for reviewers**

I refactored the test `"recovering from deleting db files while maintain connectivity"` , instead of using `time.sleep(90*time.Sceond)` for waiting for the pod restart , I used `waitForPodToFinishFullRestart` as suggested on the comments for #2104.

**- How to verify it**

in order to try it yourself you can run from the `test` directory:
```
make control-plane WHAT="make control-plane WHAT="recovering from deleting db files while maintain connectivity"
```

**- Description for the changelog**

Fixed the waiting period for delete database-files e2e-test

### CODE REVIEW

File: test/e2e/e2e.go [hunk 1]
```diff
@@ -2406,59 +2420,61 @@ var _ = ginkgo.Describe("e2e delete databases", func() {
...
-			deleteAndRestartDB(db_pod_num, DBFileNamesToDelete, containerToDeleteFrom)
-			time.Sleep(10 * time.Second)
+
+			// start the db disruption - delete the db files and delete the db-pod in order to emulate the cluster/pod restart
+
+			// Retrive the DB pod
+			dbPod, err := f.ClientSet.CoreV1().Pods(ovnNs).Get(context.Background(), db_pod_name, metav1.GetOptions{})
+			framework.ExpectNoError(err, fmt.Sprintf("unable to get pod: %s, err: %v", db_pod_name, err))
+
+			//check that all files exists on pod
```
Comment (dcbw) [inline test/e2e/e2e.go:2453]: @OmerBenHayun could you put a space after the `//` to be consistent with other comments? Thanks!

Comment (OmerBenHayun) [reply test/e2e/e2e.go]: @dcbw  added spaces after the `\\` , could you please approve the Build CI to run?
Thanks.

File: test/e2e/e2e.go [hunk 2]
```diff
@@ -2272,83 +2273,96 @@ func getNodePodCIDR(nodeName string) (string, error) {
...
+	}
+
+	// waitForPodToFinishFullRestart waits for a the pod to finish it's reset cycle and returns.
+	waitForPodToFinishFullRestart := func(f *framework.Framework, pod *v1.Pod) {
+		err := e2epod.WaitForPodCondition(f.ClientSet, pod.Namespace, pod.Name, "not ready", 5*time.Minute, testutils.PodNotReady)
+		if err != nil {
+			framework.Failf("pod %v is not arrived to not-ready state: %v", pod.Name, err)
+		}
+		// during this stage on the restarting process we can encounter "pod not found" errors.
+		// this type of errors is valid because the pod is during restart so it will have a period that it will not be available.
```
Comment (trozet) [inline test/e2e/e2e.go:2319]: "these types of errors are valid because the pod is restarting so there will be a period of time it is unavailable"

**Review by trozet — COMMENTED** [review]
minor nits, otherwise lgtm, thanks.

File: test/e2e/e2e.go [hunk 3]
```diff
@@ -2272,83 +2273,96 @@ func getNodePodCIDR(nodeName string) (string, error) {
...
-				errStr += " " + southDBFileName
-			}
-			if errStr != "" {
-				errStr = "missing db:" + errStr
-				framework.Failf(errStr)
-			}
+	fileExistsOnPod := func(f *framework.Framework, namespace string, pod *v1.Pod, file string) bool {
+		containerFlag := fmt.Sprintf("-c=%s", pod.Spec.Containers[0].Name)
+		_, err := framework.RunKubectl(ovnNs, "exec", pod.Name, containerFlag, "--", "ls", file)
+		if err == nil{
```
[the remark this answers is not in the document]
Comment (OmerBenHayun) [reply test/e2e/e2e.go:2336]: Thanks, Done

File: test/e2e/e2e.go (hunk 2 above)
Comment (OmerBenHayun) [reply test/e2e/e2e.go:2319]: Thanks, Done

**Review by trozet — APPROVED** [review]

**Pull request merged**